spinnaker_components/common/input_filtering.c

```diff
@@ -238,42 +238,54 @@ void input_filtering_get_routes(if_collection_t *filters,
                                 if_routing_table_t *routing_table,
                                 uint32_t *routes)
 {
-  // Copy in the number of routing entries
-  routing_table->n_routes = routes[0];
-  routes++;  // Advance the pointer to the first entry
-  debug("Loading %d filter routes\n", filters->n_routes);
+  filter_arg_t args[] = {{filters, routes}};
+  input_filter_build_combined_routes(
+    routing_table, sizeof(args) / sizeof(filter_arg_t), args);
+}
+
+void input_filter_build_combined_routes(
+    if_routing_table_t *routing_table,
+    unsigned int n,
+    filter_arg_t *args
+)
+{
+  // Compute the number of routing table entries
+  routing_table->n_routes = 0;
+  for (unsigned int i = 0; i < n; i++)
+  {
+    routing_table->n_routes += args[i].data[0];
```

Comment:
If args was a struct with a count member followed by a variable-length array, this would be way nicer